tests/test_url_handler.py

```diff
@@ -0,0 +1,700 @@
+    returned = url_handler.get_metadata(project)
+
+    for each in returned:
+        assert str(each) in str(expected)
+
+
+def test_get_metadata_with_valueset():
+    project = factories.ProjectFactory()
+    valset = factories.ValuesetFactory()
+    vals = [factories.ValuesetValuesFactory(valueset=valset).value for i in range(3)]
+    metadata = factories.MetadataFactory(value_sets=[valset])
+    factories.ProjectMetadataFactory(project=project, metadata=metadata)
+    returned = url_handler.get_metadata(project)
+
+    for _, value_iter in returned:
+        for value in value_iter:
+            assert value in vals
+
+
+def test_handle_metadata(rf):
+    request = rf.post(
+        '/',
+        {
+            'color': ['blue', 'other_specify'],
+            'group': ['alpha', 'other_specify'],
+            'name': ['Dan', 'Sara'],
+            'title': ['other_specify'],
+            'job': ['other_specify'],
+        }
+    )
+    posted_data = {
+        'color': ['blue', 'other_specify'],
+        'color_other': 'magenta',
+        'group': ['alpha', 'other_specify'],
+        'name': ['Dan', 'Sara'],
+        'title': 'other_specify',
+        'title_other': 'Monsieur',
+        'job': 'other_specify',
+    }
+    expected = {
+        'color': ['blue', 'magenta'],
+        'color_other': 'magenta',
+        'group': ['alpha'],
+        'name': ['Dan', 'Sara'],
+        'title': 'Monsieur',
+        'title_other': 'Monsieur',
+        # Is this really what we want returned if the 'other' attribute isn't set?
+        'job': 'other_specify',
+    }
+
+    assert url_handler.handle_metadata(request, posted_data) == expected
+
+
+@pytest.mark.parametrize('date_str', [
+    '2006-10-25',
+    '10/25/2006',
+    '10/25/06',
+    'Oct 25 2006',
+    'Oct 25, 2006',
+    '25 Oct 2006',
+    '25 Oct, 2006',
+    'October 25 2006',
+    'October 25, 2006',
+    '25 October 2006',
+    '25 October, 2006'
+])
+def test_validate_date_with_valid_dates(date_str):
+    assert url_handler.validate_date(date_str) is not None
+
+
+def test_validate_date_with_invalid_date():
+    date_str = '2006, Oct 25'
+    assert url_handler.validate_date(date_str) is None
+
+
+def test_add_url():
+    project = factories.ProjectWithMetadataFactory()
+    attribute_names = [x.name for x in project.metadata.all()]
+    value = 'some_value'
+    form_data = {
+        'url_value': 'http://www.example.com',
+        'nominator_email': 'somebody@someplace.com',
+        'nominator_name': 'John Lambda',
+        'nominator_institution': 'UNT',
+        attribute_names[0]: value
+    }
+    expected = [
+        'You have successfully nominated {0}'.format(form_data['url_value']),
+        'You have successfully added the {0} "{1}" for {2}'.format(
+            attribute_names[0], value, form_data['url_value'])
+    ]
+
+    for each in url_handler.add_url(project, form_data):
+        assert each in expected
```

Comment:
If we can't assert the expected and returned values are equal, we probably should also check the lengths are the same, because the returned list could be missing expected elements and the test would still pass.